### DISCUSSION TITLE
Does getMaxFreq is working in silicon M3?

### DISCUSSION THREAD

**Discussion by GaelAlguiar** [oshi/oshi#2757] [General]
Its not working on M3, although is working on M2

**Comment by dbwiddis** [comment]
The calculation is done here:
https://github.com/oshi/oshi/blob/545eb03e2c71f697fecc372cce257351d8186bd8/oshi-core/src/main/java/oshi/hardware/platform/mac/MacCentralProcessor.java#L349-L375

Can you find the equivalent fields in your IO Registry?  You can download [IORegistryExplorer](https://forums.developer.apple.com/forums/thread/43270), or use `ioreg` from the command line.   Try `ioreg -r -c AppleARMIODevice -n pmgr` and if that doesn't return anything, drop off the last part and just do `ioreg -r -c AppleARMIODevice` and look for anything related to voltage states.

**Comment by dbwiddis** [comment]
~Looks like a user has dumped their [ioreg](https://www.insanelymac.com/forum/topic/358515-apple-m3-macbook-pro-ioreg/) output so I'll look at that...~

Meh, broken link.

**Reply by GaelAlguiar** [reply]
what do you mean? so it is not working?

**Reply by dbwiddis** [reply]
I mean there is no file at the ilnk. But you can produce one yourself, and upload it here.

**Comment by dbwiddis** [comment]
Hey @GaelAlguiar I just got a new M3 and the frequency detection looks just fine:
```
Apple M3 Pro
 1 physical CPU package(s)
 12 physical CPU core(s) (6 performance + 6 efficiency)
 12 logical CPU(s)
Identifier: Apple Inc. Family 0x8765edea Model 0 Stepping 0
ProcessorID: 0100000c8765edea
Microarchitecture: ARM64 SoC: Everest + Sawtooth
...
Vendor Frequency: 4.1 GHz
Max Frequency: 4.1 GHz
Current Frequencies: 2.7 GHz, 2.7 GHz, 2.7 GHz, 2.7 GHz, 2.7 GHz, 2.7 GHz, 4.1 GHz, 4.1 GHz, 4.1 GHz, 4.1 GHz, 4.1 GHz, 4.1 GHz
```

**Reply by GaelAlguiar** [reply]
which code do you use? it returns -1

**Reply by dbwiddis** [reply]
I just ran `SystemInfoTest`.  See links at https://github.com/oshi/oshi?tab=readme-ov-file#usage
